config/main.py

```diff
@@ -1525,6 +1534,12 @@ def reload(db, filename, yes, load_sysinfo, no_service_restart, force, file_form
         # Get the file from user input, else take the default file /etc/sonic/config_db{NS_id}.json
         if cfg_files:
             file = cfg_files[inst+1]
+            # Save to tmpfile in case of stdin input which can only be read once
+            if file == "/dev/stdin":
+                file_input = read_json_file(file)
+                (_, tmpfname) = tempfile.mkstemp(dir="/tmp", suffix="_configReloadStdin")
```

Comment:
Look around the login inside this function, it seems one time reading of file will be enough. Let's take further effort to reorg the existing code and optimize the logic.

Reply:
[sonic-cfggen -j](https://github.com/sonic-net/sonic-utilities/blob/master/config/main.py#L1573) read it first. Then [sonic-cfggen](https://github.com/sonic-net/sonic-utilities/blob/master/config/main.py#L1624) read it second.